ircbot/commands_impl.go

```diff
@@ -69,7 +69,50 @@ func LastPlayed(e Event) error {
 	return nil
 }
 
-func StreamerQueue(e Event) error       { return nil }
+func StreamerQueue(e Event) error       {
+	// Get queue from streamer
+	songQueue, err := e.Bot.Streamer.Queue(e.Context())
+	if err != nil {
+		return err
+	}
+
+	// If the queue is empty then we're done
+	if len(songQueue) == 0 {
+		e.EchoPublic("No queue at the moment")
+		return nil
+	}
+
+	// Calculate playback time for the queue
+	var totalQueueTime time.Duration
+	for _, song := range songQueue {
+		totalQueueTime += song.Length
+	}
+
+	// Define the message strings
+	message := "{green}Queue (/r/ time: %s):{clear} %s"
+	messageJoin := " {red}|{clear} "
+
+	// Grab metadata and color green if requestable
+	onlyMetadata := make([]string, len(songQueue))
+	for i, song := range songQueue {
+		if song.IsUserRequest {
+			onlyMetadata[i] = "{green}" + song.Metadata + "{clear}"
+		} else {
+			onlyMetadata[i] = song.Metadata
+		}
+	}
+
+	// Format the message
+	message = fmt.Sprintf(message,
+		FormatPlaybackDurationHours(totalQueueTime),
+		strings.Join(onlyMetadata, messageJoin))
+
+	// Echo out the message
+	e.EchoPublic(message)
```

Comment:
the `Echo` functions do formatting for you, so the above `fmt.Sprintf` is unneeded and you can pass them directly to `EchoPublic` instead